### PR TITLE
Continue middleware chain after download.

### PR DESCRIPTION
**Pull request by richardpringle**
connect to #59

### CODE REVIEW

**Comment by richardpringle** [comment]
@loay, @raymondfeng: please review.

File: lib/storage-handler.js [hunk 1]
```diff
@@ -258,6 +258,9 @@ exports.download = function(provider, req, res, container, file, cb) {
       reader.on('error', function(err) {
         handleError(res, err);
       });
+      reader.on('end', function() {
```
Comment (raymondfeng) [inline lib/storage-handler.js]: Does the same listener apply to line 247 too?

Comment (richardpringle) [inline lib/storage-handler.js]: If I am not mistaken, the afterRemote hook is not invoked when there is an error with the remote-method right?

Comment (richardpringle) [inline lib/storage-handler.js]: fixed above

**Comment by bajtos** [comment]
While this may be a good short-term fix, we should strongly consider rewriting the method to use the new strong-remoting type "file" instead. That will also fix the swagger description of the `download` method.

See https://github.com/strongloop/loopback/issues/2063#issuecomment-189204577 and https://github.com/strongloop/strong-remoting/pull/284 for more details.

**Comment by loay** [comment]
LGTM

**Pull request merged**